### PR TITLE
[IMPROVE] Quotes on E2EE Messages

### PR DESCRIPTION
**Pull request by yash-rajpal**
## Proposed changes (including videos or screenshots)

This PR provides basic quoting to E2EE messages, it doesn't support multiple chaining of quotes messages.

Screenshot 2022-07-19 at 9 42 19 PM

## Issue(s)

#19110 
#16148 
## Further comments

This implementation has a few limitations on how quoting will work in E2EE messages. Quotes are handled as attachments and generally quotes are added by the server when it receives the message with some message-link in its content. This approach is not possible for E2EE messages as server receives the encrypted message and cannot get the links. So quotes on E2EE messages have to be handled on Client side.

Here we use the API `chat.getMessage` to get the quoted message so we cannot see quoted messages of rooms of which user is not a part of.

We can also not chain multiple quote messages as server has no information on existing quote attachments on a E2EE message (only client has that info), though we can handle this on client side meteor collections, but not sure if that is a good idea.

### CODE REVIEW

File: apps/meteor/app/e2e/client/rocketchat.e2e.js [hunk 1]
```diff
@@ -402,6 +411,64 @@ class E2E extends Emitter {
...
+
+			if (!getQuotedMessage?.success || !quotedMessage) {
+				return;
+			}
+
+			const decryptedQuoteMessage = await this.decryptMessage(quotedMessage);
+
+			message.attachments = message.attachments || [];
+
+			const quoteAttachment = {
```
Comment (gabriellsh) [inline apps/meteor/app/e2e/client/rocketchat.e2e.js:447]: I think we do something similar in backend for regular messages, right? Is there any possibility we could implement a function to generate an attachment from a message and use it in both places? We could event unit test this function after...

**Review by gabriellsh — CHANGES_REQUESTED** [review]
I'd like to see e2e-ui tests for this too.

**Review by gabriellsh — APPROVED** [review]

File: apps/meteor/app/e2e/client/rocketchat.e2e.ts [hunk 2]
```diff
@@ -370,38 +406,86 @@ class E2E extends Emitter {
...
 		}
 
-		return {
+		const decryptedMessage: IE2EEMessage = {
 			...message,
 			msg: data.text,
 			e2e: 'done',
 		};
+
+		console.log('before quote', decryptedMessage);
```
Comment (sampaiodiego) [inline apps/meteor/app/e2e/client/rocketchat.e2e.ts:415]: oopsie.. can someone pls remove these? 🙈

Comment (yash-rajpal) [reply apps/meteor/app/e2e/client/rocketchat.e2e.ts:415]: Removed here - https://github.com/RocketChat/Rocket.Chat/pull/27092/commits/2d4f247bce23cf3e9728caab348665eb58a3bebb

Thanks!

**Comment by KamilSucharski** [comment]
@yash-rajpal You linked this issue for this PR https://github.com/RocketChat/Rocket.Chat/issues/24721 , but it also requires working attachment previews in E2E to be closed. Is that included, or did you only modify quoting?

**Pull request merged**